lib/SILGen/SILGenType.cpp

```diff
@@ -613,6 +613,20 @@ SILFunction *SILGenModule::emitProtocolWitness(
   auto input = reqtOrigTy->getInput().subst(reqtSubMap)->getCanonicalType();
   auto result = reqtOrigTy->getResult().subst(reqtSubMap)->getCanonicalType();
 
+  // If there's something to map to for the witness thunk, the conformance
+  // should be phrased in the same terms. This particularly applies to classes
+  // where a thunk for a method in a conformance like `extension Class: P where
+  // T: Q` will go from its native signature of `<τ_0_0 where τ_0_0: Q>` (with T
+  // canonicalised to τ_0_0), to `<τ_0_0, τ_1_0 where τ_0_0: Class<τ_1_0>,
+  // τ_1_0: Q>` (with T now represented by τ_1_0). Find the right conformance by
+  // looking for the conformance of 'Self'.
+  if (reqtSubMap) {
+    auto requirement = conformance.getRequirement();
+    auto self = requirement->getProtocolSelfType()->getCanonicalType();
+
+    conformance = *reqtSubMap.lookupConformance(self, requirement);
```

Comment:
I suspect this is the same as conformance.subst(reqtSubMap), which eliminates the need for the conditional, but it doesn't matter.

Reply:
Unfortunately I don't think so; it looks like `reqtSubMap` is for substituting in the protocols context, replacing `Self` rather than the parameters of the generic type.